api/src/routes/users.ts

```diff
@@ -98,4 +103,46 @@ router.delete("/connections/:id", routeProtector, [ param("id").not().isEmpty().
   }
 });
 
+
+router.patch("/:id", routeProtector, [body("firstName").trim().escape(),
+body("lastName").trim().escape(),
+body("avatar").isURL().trim(),
```

Comment:
Can we either make each item to update optional or make a separate route for updating avatar. The profile editor will only edit names and job title. Avatar will be a separate event.

Reply:
> Can we either make each item to update optional or make a separate route for updating avatar. The profile editor will only edit names and job title. Avatar will be a separate event.

I believe it's already setup for each element to be optional. It will only update what is received from the request. So you should be able to send a request to this route with just the avatarUrl

Reply:
Okay thanks, I'll try it some more

Reply:
Hey so this doesn't work, if I exclude avatar or set it as "" it returns a 400, but does work if I use a valid URL

Reply:
It's fine in the model but I think the validation middle where is expecting it.

Reply:
Updated: Hey I fixed this using custom validator. Thanks for testing this!